keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -413,28 +420,179 @@ On Pod Create/Update/Delete:
 
 ### Kube-Proxy
 
-Kube-proxy will be modified to consume EndpointSlice instances besides Endpoints resource. A flag will be added to kube-proxy to toggle the mode. 
-
-```
-Watch: Service, EndpointSlice ==> Manage: iptables, ipvs, etc
-```
-- Merge multiple EndpointSlice into an aggregated list.
-- Reuse the existing processing logic 
-
-### Endpoint Controller (classic)
-
-In order to ensure backward compatibility for external consumer of the core/v1 Endpoints API, the existing K8s endpoint controller will keep running until the API is EOL. The following limitations will apply:
-- Starting from EndpointSlice beta: If # of endpoints in one Endpoints object exceed 1000, generate a warning event to the object. 
-- Starting from EndpointSlice GA: Only include up to 1000 endpoints in one Endpoints Object and throw events.
+Kube-Proxy will support consuming EndpointSlice resources as an alternative to
+Endpoints resources. This will be enabled with the `EndpointSliceProxying`
+feature gate.
+
+This will require minimal changes to proxier implementations, instead updating
+the shared code that transforms Endpoints into a generic data structure to also
+transform EndpointSlices into that same data structure.
+
+### Endpoints Controller (classic)
+
+In order to ensure backwards compatibility for external consumer of the core/v1
+Endpoints API, the existing Endpoints controller will keep running until the
+API is EOL. After EndpointSlices become GA, the Endpoints controller will
+gradually limit functionality.
+
+* Kubernetes 1.20: If the number of endpoints in one Endpoints object exceeds
+  1000, set `endpoints.kubernetes.io/over-capacity` label to "warning".
+* Kubernetes 1.21: Limit the number of endpoints in one Endpoints object to 1000
+  and set the `endpoints.kubernetes.io/over-capacity` label to "truncated" when
+  truncation occurs.
+
+### EndpointSliceMirroring Controller
+
+In some cases, custom Endpoints resources are created by applications. To ensure
+that these applications will not need to concurrently write to both Endpoints
+and EndpointSlice resources, a new EndpointSliceMirroring controller will be
+used to mirror custom Endpoints resources to corresponding EndpointSlices.
+
+* A new `endpointslice.kubernetes.io/skip-mirror` label will be introduced.
```

Comment:
Can't we do this without a label?

If not, it's a bit weird to have an `endpointslice.kubernetes.io` label on an Endpoints.  Maybe `endpoints.kubernetes.io/disable-endpoint-slice-mirror` ?  Minor point